pkg/controller/master/addon/addon_processing.go

```diff
@@ -0,0 +1,301 @@
+	// chart exists and helmChart deletion has been triggered
+	// wait for chart job to be completed
+	if !hc.DeletionTimestamp.IsZero() {
+		// if deletion job has not been set then requeue and retry again
+		if !strings.Contains(hc.Status.JobName, helmChartJobDeletePrefix) {
+			return h.enqueueAfter(aObj)
+		}
+		j, ok, err := h.currentDeletionJob(hc)
+		if err != nil {
+			return aObj, err
+		}
+
+		if ok {
+			// if job is not complete, wait for completion
+			if !isJobComplete(j) {
+				return h.enqueueAfter(aObj)
+			}
+
+			if j.Status.Failed > 0 {
+				markErrorCondition(aObj, fmt.Errorf("addon disable job %s failed", j.Name))
+			}
+
+			if j.Status.Succeeded > 0 {
+				markCompletedCondition(aObj)
+				aObj.Status.Status = harvesterv1.AddonDisabled
+			}
+
+			h.addon.UpdateStatus(aObj)
+		}
+
+	}
+
+	return aObj, nil
+}
+
+// enableAddon will check for addon status and create a helmChart if needed and setup
+// appropriate conditions to ensure that addon - helmchart state can reconcile
+func (h *Handler) processEnableAddon(a *harvesterv1.Addon) (*harvesterv1.Addon, error) {
+	aObj := a.DeepCopy()
+
+	// since re-enable of an Addon is possible from AddonDisabling state after a failure
+	// we need to ensure the underlying HC has been manually cleared up
+	hc, owned, err := h.getAddonHelmChart(aObj)
+	if err != nil {
+		return aObj, err
+	}
+
+	if owned && hc != nil && !hc.DeletionTimestamp.IsZero() {
+		logrus.Warnf("helmChart %s has a deletion timestamp. likely from an older install, wait for GC before re-installing", hc.Name)
+		return h.enqueueAfter(aObj)
+	}
+
+	if !owned && hc != nil {
+		logrus.Warnf("cannot create helmChart %s, as another un-owned chart with same name already exists. requeuing until this is removed", hc.Name)
+	}
+	// ensure condition is set first. This is to ensure the timestamp is before helm chart creation
+	// which allows us to be certain that the job creation timestamp should be after this condition
+	if !harvesterv1.AddonOperationInProgress.IsTrue(aObj) {
+		markInProgressCondition(aObj)
+		return h.addon.UpdateStatus(aObj)
+	}
+
+	if hc == nil {
+		if err := h.deployHelmChart(aObj); err != nil {
+			return aObj, fmt.Errorf("error deploying helmChart: %v", err)
+		}
+	}
+
+	aObj.Status.Status = harvesterv1.AddonEnabling
+	return h.addon.UpdateStatus(aObj)
+}
+
+// processEnablingAddon will check if the job for deploying the helmChart is completed
+// based on status of job, the addon deployment will be marked successful or errored
+// and status will be updated accordingly
+func (h *Handler) processEnablingAddon(a *harvesterv1.Addon) (*harvesterv1.Addon, error) {
+	aObj := a.DeepCopy()
+	hc, _, err := h.getAddonHelmChart(aObj)
+	if err != nil {
+		return aObj, err
+	}
+
+	if hc == nil {
+		// unable to find helmChart. Likely manually GC'd. Reset to initial state
+		// this will allow addon to be processed again and redeployed
+		markCompletedCondition(aObj)
+		aObj.Status.Status = harvesterv1.AddonDisabled
```

Comment:
that is the aim. if helm chart object is not found, because of manual cleanup, then resetting the status allows the addon to fall through the processing loop again.

This will result in the helm chart being created again since the spec is enabled.